### PR TITLE
restore the separate vectors of mtime and filehash in file_storage

### PR DESCRIPTION
**Pull request by arvidn**

### CODE REVIEW

File: src/file_storage.cpp [hunk 1]
```diff
@@ -1048,8 +1027,8 @@ namespace {
 	std::time_t file_storage::mtime(internal_file_entry const& fe) const
 	{
 		int index = int(&fe - &m_files.front());
-		TORRENT_ASSERT_PRECOND(index >= 0 && index < int(m_files.size()));
```
Comment (arvidn) [inline src/file_storage.cpp:1051]: actually, these asserts should probably be preserved

File: src/file_storage.cpp [hunk 2]
```diff
@@ -741,10 +712,13 @@ namespace {
 		e.hidden_attribute = bool(file_flags & file_storage::flag_hidden);
 		e.executable_attribute = bool(file_flags & file_storage::flag_executable);
 		e.symlink_attribute = bool(file_flags & file_storage::flag_symlink);
-		e.hash = filehash;
 		e.root = root_hash;
-		e.mtime = mtime;
 
+		if (filehash)
+		{
+			if (m_file_hashes.size() < m_files.size()) m_file_hashes.resize(m_files.size());
```
Comment (ssiloti) [inline src/file_storage.cpp:719]: The new elements in the vector need to be initialized to `nullptr`. Same goes for `m_time` below. And actually now that I look at it this issue is also in master/RC_1_2/RC_1_1.

Comment (arvidn) [reply src/file_storage.cpp:719]: I'm pretty sure `std::vector` always default initialize elements, which will mean `0` and `nullptr` in this case

Comment (arvidn) [reply src/file_storage.cpp:719]: and by "pretty sure" I really mean "confident". https://en.cppreference.com/w/cpp/container/vector/resize

**Pull request merged**